irohad/consensus/yac/storage/impl/buffered_cleanup_strategy.cpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/yac/storage/buffered_cleanup_strategy.hpp"
+
+#include "common/visitor.hpp"
+
+using namespace iroha::consensus::yac;
+
+boost::optional<CleanupStrategy::RoundsType> BufferedCleanupStrategy::finalize(
```

Comment:
Why not just return a container (possibly empty)? This will make the code more general.

Reply:
My guess is that this case is similar to `optional<shared_ptr<T>>`, where `shared_ptr` is guaranteed not to be `nullptr`. It is probably guaranteed here that if container is returned, it is not empty.

Reply:
IMHO it only complicates the code. Moreover, containers are mostly iterated in a loop, and not dereferenced like pointers. And we even have the empty vector constructed in the function.

Reply:
I do not think one should safely do `if(opt_container) { anything(opt_container[0]); }`. And for indices higher than `0` you must anyway check the size.

Reply:
There were two points for doing that: first, as Andrei said, second, use bind for in vote storage,
```
strategy->finalize(round, outcome) | ...
```